corehq/blobs/tests/test_util.py

```diff
@@ -16,3 +19,32 @@ def test_random_id_length(self):
 
     def test_random_id_randomness(self):
         self.assertEqual(len(set(self.ids)), self.sample_size, self.ids)
+
+
+class TestGzipCompressReadStream(TestCase):
+
+    def _is_gzip_compressed(self, file_):
+        with gzip.open(file_, 'r') as f:
+            try:
+                f.read(1)
+                return True
+            except OSError:
+                return False
+
+    def test_compression(self):
+        with tempfile.NamedTemporaryFile() as f:
+            f.write(b"x")
+            compress_stream = mod.GzipCompressReadStream(f)
+            with tempfile.NamedTemporaryFile() as compressed_f:
+                compressed_f.write(compress_stream.read())
+                self.assertTrue(self._is_gzip_compressed(compressed_f))
+
+    def test_content_length(self):
+        with tempfile.NamedTemporaryFile() as f:
+            f.write(b"x")
+            compress_stream = mod.GzipCompressReadStream(f)
+            with self.assertRaises(Exception):
+                compress_stream.content_length
+            content_length = len(compress_stream.read())
+            self.assertEqual(compress_stream.content_length, content_length)
+
```

Comment:
Please add another test: write several bytes, then do a partial read (say write 10 bytes, then read 5), verify that exception is raised on access `compress_stream.content_length` before all 10 bytes have been read.